[
    {
        "subtype": "channel_join",
        "user": "UG9FZ8RBP",
        "text": "<@UG9FZ8RBP> has joined the channel",
        "type": "message",
        "ts": "1550399193.054800"
    },
    {
        "subtype": "channel_join",
        "user": "UGAAAK8R5",
        "text": "<@UGAAAK8R5> has joined the channel",
        "type": "message",
        "ts": "1550415574.055000"
    },
    {
        "user": "UFTBRB3F1",
        "type": "message",
        "ts": "1550417321.059300",
        "client_msg_id": "ccf12c58-c3e5-4ada-97b8-d99549fc4578",
        "text": "I have got that sample code up and running, but for me it behaves the same as the previous effort (almost), in that it sends through the initial price cache as expected, however then it just pings away (i.e. the client console.log shows 'pinging').\n\nThe server logs, show:\n\n```\nINFO:socketio.server:received event \"ping\" from 41f1cafd420740b3921c67883d9b0bb0 [\/]\nINFO:socketio.server:emitting event \"my_response\" to 41f1cafd420740b3921c67883d9b0bb0 [\/]\nINFO:engineio.server:41f1cafd420740b3921c67883d9b0bb0: Client is gone, closing socket\nINFO:engineio.server:41f1cafd420740b3921c67883d9b0bb0: Client is gone, closing socket\n```\n\nThe client logs are interesting, as there are some errors that show up (which we never had previously):\n\n```\nsocket.io.min.js:1 POST <http:\/\/localhost:5000\/socket.io\/?EIO=3&amp;transport=polling&amp;t=1550416830901-24&amp;sid=0e8737cace6149e1abff289ba77e7c24> 500 (Internal Server Error)\nsocket.io.min.js:1 GET <http:\/\/localhost:5000\/socket.io\/?EIO=3&amp;transport=polling&amp;t=1550416745904-23&amp;sid=0e8737cace6149e1abff289ba77e7c24> 400 (BAD REQUEST)\n```\n\nand also:\n\n```\nsocket.io.min.js:2 WebSocket connection to '<ws:\/\/localhost:5000\/socket.io\/?EIO=3&amp;transport=websocket&amp;sid=9ae605a3067b40328159b3072a45112c>' failed: WebSocket is closed before the connection is established.\n```",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g771e110be94",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/a771e110be945b86bede9d5c1bb551e3.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0024-72.png",
            "first_name": "",
            "real_name": "Newbie99",
            "display_name": "",
            "team": "T4G9NBD2M",
            "name": "vjmehra",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "kzZTp",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "I have got that sample code up and running, but for me it behaves the same as the previous effort (almost), in that it sends through the initial price cache as expected, however then it just pings away (i.e. the client console.log shows 'pinging').\n\nThe server logs, show:\n\n"
                            }
                        ]
                    },
                    {
                        "type": "rich_text_preformatted",
                        "elements": [
                            {
                                "type": "text",
                                "text": "INFO:socketio.server:received event \"ping\" from 41f1cafd420740b3921c67883d9b0bb0 [\/]\nINFO:socketio.server:emitting event \"my_response\" to 41f1cafd420740b3921c67883d9b0bb0 [\/]\nINFO:engineio.server:41f1cafd420740b3921c67883d9b0bb0: Client is gone, closing socket\nINFO:engineio.server:41f1cafd420740b3921c67883d9b0bb0: Client is gone, closing socket\n"
                            }
                        ]
                    },
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "\n\nThe client logs are interesting, as there are some errors that show up (which we never had previously):\n\n"
                            }
                        ]
                    },
                    {
                        "type": "rich_text_preformatted",
                        "elements": [
                            {
                                "type": "text",
                                "text": "socket.io.min.js:1 POST "
                            },
                            {
                                "type": "link",
                                "url": "http:\/\/localhost:5000\/socket.io\/?EIO=3&amp;transport=polling&amp;t=1550416830901-24&amp;sid=0e8737cace6149e1abff289ba77e7c24",
                                "text": "http:\/\/localhost:5000\/socket.io\/?EIO=3&transport=polling&t=1550416830901-24&sid=0e8737cace6149e1abff289ba77e7c24"
                            },
                            {
                                "type": "text",
                                "text": " 500 (Internal Server Error)\nsocket.io.min.js:1 GET "
                            },
                            {
                                "type": "link",
                                "url": "http:\/\/localhost:5000\/socket.io\/?EIO=3&amp;transport=polling&amp;t=1550416745904-23&amp;sid=0e8737cace6149e1abff289ba77e7c24",
                                "text": "http:\/\/localhost:5000\/socket.io\/?EIO=3&transport=polling&t=1550416745904-23&sid=0e8737cace6149e1abff289ba77e7c24"
                            },
                            {
                                "type": "text",
                                "text": " 400 (BAD REQUEST)\n"
                            }
                        ]
                    },
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "\n\nand also:\n\n"
                            }
                        ]
                    },
                    {
                        "type": "rich_text_preformatted",
                        "elements": [
                            {
                                "type": "text",
                                "text": "socket.io.min.js:2 WebSocket connection to '"
                            },
                            {
                                "type": "link",
                                "url": "ws:\/\/localhost:5000\/socket.io\/?EIO=3&amp;transport=websocket&amp;sid=9ae605a3067b40328159b3072a45112c",
                                "text": "ws:\/\/localhost:5000\/socket.io\/?EIO=3&transport=websocket&sid=9ae605a3067b40328159b3072a45112c"
                            },
                            {
                                "type": "text",
                                "text": "' failed: WebSocket is closed before the connection is established.\n"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UFTBRB3F1",
        "type": "message",
        "ts": "1550425894.061300",
        "client_msg_id": "d287598f-3098-44b9-a140-8037d37362da",
        "text": "By emitting a text string and commenting out the market_books references below, I can see the PING and respective emit work correctly, therefore this issue appears to be around market_books = output_queue.get() after the first time (as it always works first time round):\n\n```\n@socketio.on('ping')\ndef handle_message(*_args, **_kwargs):\n    market_books = output_queue.get()\n\n    for market_book in market_books:\n        emit('my_response',\n             {'data': 'update', 'mb': market_book.streaming_update})\n```",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g771e110be94",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/a771e110be945b86bede9d5c1bb551e3.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0024-72.png",
            "first_name": "",
            "real_name": "Newbie99",
            "display_name": "",
            "team": "T4G9NBD2M",
            "name": "vjmehra",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "rMjf3",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "By emitting a text string and commenting out the market_books references below, I can see the PING and respective emit work correctly, therefore this issue appears to be around market_books = output_queue.get() after the first time (as it always works first time round):\n\n"
                            }
                        ]
                    },
                    {
                        "type": "rich_text_preformatted",
                        "elements": [
                            {
                                "type": "text",
                                "text": "@socketio.on('ping')\ndef handle_message(*_args, **_kwargs):\n    market_books = output_queue.get()\n\n    for market_book in market_books:\n        emit('my_response',\n             {'data': 'update', 'mb': market_book.streaming_update})\n"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "subtype": "channel_join",
        "user": "UG99A16NQ",
        "text": "<@UG99A16NQ> has joined the channel",
        "type": "message",
        "ts": "1550433623.061500"
    }
]